### ISSUE TITLE
BUG: SC1073 triggered on valid code and shellcheck stops processing

### ISSUE THREAD

**Issue by r6d2** [koalaman/shellcheck#1268]
#### For bugs
- Rule Id: SC1073,SC1019,SC1020,SC1072
- My shellcheck version: online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env bash
declare -A ARRAY=([alpha]=a [bravo]=b)
IDX=--alpha
[[ -v ARRAY[${IDX#--}] ]] && echo "ARRAY[${IDX#--}]=${ARRAY[${IDX#--}]}"

```

#### Here's what shellcheck currently says:
```
Line 4	SC1073: Couldn't parse this test expression. Fix to allow more checks.
Line 4	SC1019: Expected this to be an argument to the unary condition.
Line 4	SC1020: You need a space before the ]].
Line 4	SC1072: Missing space before ]. Fix any mentioned problems and try again.
```
#### Here's what I wanted or expected to see:

Nothing. AFAIK the code is valid and executes with no issues. This is the output with ```xtrace``` enabled:
```
+ ARRAY=([alpha]=a [bravo]=b)
+ declare -A ARRAY
+ IDX=--alpha
+ [[ -v ARRAY[alpha] ]]
+ echo 'ARRAY[alpha]=a'
ARRAY[alpha]=a
```

I think the problem relies on shellcheck not being able to recognize the parameter substitution used on the array subscript of the ```[[ -v ... ]]``` test. If I edit the condition to ```[[ -v ARRAY[$IDX] ]]``` I get the ```No issues detected!``` happy ending :-) This breaks the code, of course.

Unfortunately, this issue is critical for me. Shellcheck refuses to process anything else until I "fix" my code, and no directives disabling the reported conditions allow shellcheck to proceed.

Any suggestions, other than selectively commenting the "offending" lines of code?

**Comment by koalaman** [comment]
Fixed by @ngzhian. Thanks!
